src/mod/server.mod/servmsg.c

```diff
@@ -1132,6 +1141,115 @@ static int got311(char *from, char *msg)
   return 0;
 }
 
+static int gotauthenticate(char *from, char *msg)
+{
+  char src[256] = ""; // FIXME: size
+  char *s;
+  size_t slen;
+  unsigned char dst[256] = ""; // FIXME: size
+  size_t olen;
+  unsigned char dst2[256] = ""; // FIXME: size
+  unsigned int olen2;
+#ifdef HAVE_OPENSSL_SSL_H
+  FILE *fp;
+  EC_KEY *eckey;
+  EVP_PKEY *privateKey;
+#endif
+
+  putlog(LOG_SERV, "*", "SASL: got AUTHENTICATE %s", msg);
+  if (msg[0] == '+') {
+    s = src;
+    /* Don't use snprintf due to \0 inside */
+    if (!strcmp(sasl_mechanism, "PLAIN")) {
+      strcpy(s, sasl_username);
+      s += strlen(sasl_username) + 1;
+      strcpy(s, sasl_username);
+      s += strlen(sasl_username) + 1;
+      strcpy(s, sasl_password);
+      s += strlen(sasl_password);
+    }
+    else if (!strcmp(sasl_mechanism, "ECDSA-NIST256P-CHALLENGE")) {
+      strcpy(s, sasl_username);
+      s += strlen(sasl_username) + 1;
+      strcpy(s, sasl_username);
+      s += strlen(sasl_username);
+    }
+    else /* sasl_mechanism == EXTERNAL */
```

Comment:
Maybe run this as an else if as well, and if something like '4' is added, fail?

Reply:
you made the same valid point as grawity. ill leave it as TODO for now.